tensorflow/lite/kernels/internal/common.h

```diff
@@ -230,14 +230,18 @@ template <typename T>
 int CountLeadingZeros(T integer_input) {
   static_assert(std::is_unsigned<T>::value,
                 "Only unsigned integer types handled.");
-#if defined(__GNUC__)
-  return integer_input ? __builtin_clz(integer_input)
-                       : std::numeric_limits<T>::digits;
-#else
   if (integer_input == 0) {
     return std::numeric_limits<T>::digits;
   }
-
+#if defined(__GNUC__)
+  if (std::is_same<T, unsigned long long>::value) {
```

Comment:
please use uint64_t and uint32_t instead of unsinged long long and unsinged long, thanks

Reply:
As `__builtin_clz`, `__builtin_clzl` and `__builtin_clzll` respectively take an `unsigned int`, `unsigned long` and `unsigned long long` as parameters, shouldn't it be better to keep these? Or was there a compilation warning?

We could use `uint32_t` for `int` (assuming that all platforms TFLite supports has a 32-bit int) and `uint64_t` for `unsigned long long` and remove `unsigned long` as I think its size varies on some of the TFLite supported platforms.

Reply:
we are getting lint issue with the unsigned uint and unsigned long

maybe we can do things like here: https://github.com/tensorflow/tensorflow/blob/master/tensorflow/lite/experimental/microfrontend/lib/bits.h#L64-L92

Reply:
I removed the `unsigned long` as it is usually equal to an `unsigned int` or `unsigned long long` depending on the platform. Would something like https://github.com/tensorflow/tensorflow/pull/52014/files#diff-12b8b526d5fda290ee64fffd6db1c8b7908d6b375a79ac08335f9b6a76ff72ffL230 be alright? It allows us to also support type smaller than an `int`.